fs/filter/filter.go

```diff
@@ -192,6 +192,10 @@ func NewFilter(opt *Opt) (f *Filter, err error) {
 		}
 	}
 	for _, rule := range f.Opt.FilesFrom {
+		if !isFilesFromOptUsedAlone(f.Opt) {
```

Comment:
I think you could simplify this a bit by using `f.InActive()` to check if anything had been added to the filter, then you wouldn't need the `isFilesFromOptUsedAlone` function.

Something like

```go
inActive := f.InActive()
for _, rule := range f.Opt.FilesFrom {
    if !inActive {
        return nil, fmt.Errorf("...")
```

Reply:
Hi @ncw, you're very welcome and thanks for your review and advice.

I must confess a bit of initial skepticism for replacing my function with the InActive method because with a raw comparison of the two functions I was fearing that some Opt(s) might have not been checked.

Not a situation I would have expected: I trust, of course, more your suggestion than my raw comparison, but to make things completely straight I added an additional test, that passed with the code change you propose.

This is the test:

```go
func TestNewFilterForbiddenMixOfFilesFromAndFilterRule(t *testing.T) {
	Opt := DefaultOpt

	// Set up the input
	Opt.FilterRule = []string{"- filter1", "- filter1b"}
	Opt.FilesFrom = []string{testFile(t, "#comment\nfiles1\nfiles2\n")}

	rm := func(p string) {
		err := os.Remove(p)
		if err != nil {
			t.Logf("error removing %q: %v", p, err)
		}
	}
	// Reset the input
	defer func() {
		rm(Opt.FilesFrom[0])
	}()

	_, err := NewFilter(&Opt)
	require.Error(t, err)
}
```

In conclusion, I totally approve your suggestion and I ask you whether you're ok that I also add this test case or not. In general I would say yes, but maybe you reckon that the new `TestNewFilterFull` and `TestNewFilterFullExceptFilesFromOpt` already cover this situation sufficiently.

What's your view on this?

Reply:
Alright, thank you for your advice. I have just committed the change and the new test case